pkg/controllers/machine_controller.go

```diff
@@ -806,6 +811,38 @@ func (r *MachineReconciler) setTCMallocPath(domain *libvirtxml.Domain) error {
 	return nil
 }
 
+func (r *MachineReconciler) setGuestAgent(machine *api.Machine, domainDesc *libvirtxml.Domain) {
+	if machine.Spec.GuestAgent == api.GuestAgentNone {
```

Comment:
Can we have this change outside, so that we know code doesn't enter at all if it's disabled? Just a cosmetic suggestion.

Reply:
how do you mean that? I prefer have all logic in function. It is logical block, so you have to just delete this function and calls to it and you don't need study code what else you have to delete.

Reply:
I would usually prefer to keep the optional features behind the `if`, then we know code-flow never enters into this method when flag is set, and main flow is safe-guarded. 
In this case, it's an optional suggestion, feel free to resolve it otherwise. 
```golang
if conditionMet {
   setGuestAgent()
}
```

Reply:
fixed